extensions/file-manager/src/manage-files.tsx

```diff
@@ -1,5 +1,36 @@
-import { getStartDirectory, Directory } from "./utils";
+import { useCallback, useRef, useState } from "react";
+import { Directory } from "./ui/Directory";
+import { getStartDirectory } from "./utils";
+import { AppContext, AppContextType } from "./utils/AppContext";
 
 export default function Command() {
-  return <Directory path={getStartDirectory()} />;
+  const [pathInContext, setPathInContext] = useState("");
+
+  const refShouldReRender = useRef(false);
```

Comment:
I used the React ref here so when callers use `markFinishRenderingDirectory` function, it does not trigger a new rendering.